src/raft.rs

```diff
@@ -545,41 +546,159 @@ impl<T: Storage> Raft<T> {
         is_batched
     }
 
+    // Whether the given peer could use Follower Replication
+    fn use_delegate(&self, to: u64) -> bool {
```

Comment:
I prefer
```rust
fn use_delegate(&self, to: u64) -> bool {
    self.is_leader() &&
        self.groups.get_members(to).map_or(0, |members| members.len()) > 1 &&
        !self.groups.in_same_group(self.id, to)
}
```
And we need to make sure that peers with invalid group_id shouldn't exist in `groups`.